AppSettingsManagerAPI/AppSettingsManagerApi.Infrastructure/MySql/PermissionsManager.cs

```diff
@@ -0,0 +1,75 @@
+// The PermissionsManager class implements all the methods needed to 
+// check whether the user has the necessary permissions to access a settingGroup
+// and allows the user to request permission to a settingGroup if already don't have it
+// The user (owner of the project or admin) can then approve or reject the request
+public class PermissionsManager
+{
+    private readonly List<Permission> _permissions;
+
+    public PermissionsManager()
+    {
+        _permissions = new List<Permission>();
+    }
+
+    public void AddPermission(Permission permission)
+    {
+        _permissions.Add(permission);
+    }
```

Comment:
```suggestion
    public async Task<Permission> AddPermission(AddPermissionRequest request)
    {
        var permission = new Permission {
            // construct new permission object from request object
        }
        
        _settingsContext.Permissions.Add(permission);
        await _settingsContext.SaveChangesAsync();
        
        return _permissionConverter.Convert(permission);
    }
```

This might seem tricky but it's almost the same as the methods we've already implemented for Users and Settings